### ISSUE TITLE
Grouped legend alignment issue from 1.50 

### ISSUE THREAD

**Issue by mjordan84** [plotly/plotly.js#4336]
Hi,

I've been using plotly.js for a few month now (great work by the way) and noticed that since 1.50 legend groups are no longer aligned. As you can see in the attached example the 5th group is not aligned with the 2nd. 
Is this an expected behaviour?

Example: 
https://codepen.io/mjordan84/pen/LYYdeog 

Thanks,
Marc

**Comment by archmoj** [comment]
Possibly related to https://github.com/plotly/plotly.js/pull/4160.

**Comment by etpinard** [comment]
That's correct, the behaviour changed after #4160 (released in `v1.50.0`). 

I didn't intend to change it (see pre-1.50.0 behaviour here: https://codepen.io/etpinard/pen/abbXKpM), but I can't say which behaviour is _better_. What do you think @mjordan84 and @archmoj ?

That said, a breaking change is a breaking change, so I'll tag this as `type: bug`. If some people like the current (post `v1.50.0`) behaviour better, maybe we should think about adding a new grouped legend option.

This ticket is somewhat related to https://github.com/plotly/plotly.js/issues/960

**Comment by etpinard** [comment]
current:

![image](https://user-images.githubusercontent.com/6675409/69184012-04f0d480-0ae2-11ea-945d-e26ce99e518e.png)


pre `v1.50.0`

![image](https://user-images.githubusercontent.com/6675409/69184046-16d27780-0ae2-11ea-9985-64882781e2ec.png)

**Comment by mjordan84** [comment]
@etpinard as you said both seems valid. The new behaviour is a bit more problematic when only the first Item of each legend group is visible.
![image](https://user-images.githubusercontent.com/55309143/69191567-dffa7100-0b76-11ea-8ac0-b828b82cd489.png)

**Comment by etpinard** [comment]
> The new behaviour is a bit more problematic when only the first Item of each legend group is visible.

You mean when all but the first trace of each group has `showlegend: false`?

**Comment by mjordan84** [comment]
Yes @etpinard , when the first element of each group has `showlegend: true` then the rest set to `false` in that group.
https://codepen.io/mjordan84/pen/GRRzXoz

**Comment by gvwilson** [comment]
Hi - we are trying to tidy up the stale issues and PRs in Plotly's public repositories so that we can focus on things that are still important to our community. Since this one has been sitting for several years, I'm going to close it; if it is still a concern, please add a comment letting us know what recent version of our software you've checked it with so that I can reopen it and add it to our backlog. Thanks for your help - @gvwilson

**Comment by mjordan84** [comment]
The issue is still present in the latest version.

**Comment by gvwilson** [comment]
Thanks @mjordan84 - I'll reopen it and try to get it into our backlog
